src/main/java/core/basesyntax/service/DataConverterImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line: data) {
+            String[] elements = line.split(",");
+            FruitTransaction fruitTransaction =
+                    new FruitTransaction(FruitTransaction.Operation.fromCode(elements[0]),
+                            elements[1], Integer.parseInt(elements[2]));
```

Comment:
Consider adding validation or error handling for the input data to ensure that `elements` has the expected number of parts and that `elements[2]` can be parsed as an integer. This will help prevent potential `ArrayIndexOutOfBoundsException` or `NumberFormatException` if the input data is malformed.